### PR TITLE
com.darhon.syncbackup

### PR DESCRIPTION
**Pull request by DarhonSoftware**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.darhon.syncbackup.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+app-id: com.darhon.syncbackup
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

File: com.darhon.syncbackup.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  - --filesystem=host
+modules:
+  - name: rsync
+    buildsystem: autotools
+    config-opts:
+      - --disable-zstd
+      - --disable-xxhash
+    sources:
+      - type: archive
+        path: rsync_3.2.3.orig.tar.gz #https://packages.ubuntu.com/source/impish/rsync
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:23]: you need a URL here.

File: com.darhon.syncbackup.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    cleanup:
+      - /share/man
+
+  - name: ksshaskpass
+    buildsystem: cmake
+    sources:
+      - type: archive
+        path: ksshaskpass_5.22.5.orig.tar.xz #https://packages.ubuntu.com/source/impish/ksshaskpass
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:31]: you need an actual URL here.

File: com.darhon.syncbackup.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    cleanup:
+      - /share/man
+
+  - name: syncbackup
+    buildsystem: cmake
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:38]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

File: com.darhon.syncbackup.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    sources:
+      - type: archive
+        path: syncbackup_2.0.tar.xz
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:41]: you need an URL here.

File: com.darhon.syncbackup.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+#    buildsystem: simple
+#    build-commands:
+#      - install -D syncbackup.sh $FLATPAK_DEST/bin/syncbackup.sh
+#    sources:
+#      - type: script
+#        dest-filename: syncbackup.sh
+#        commands: 
+#          - export SSH_ASKPASS=$FLATPAK_DEST/bin/ksshaskpass
+#          - $FLATPAK_DEST/bin/syncbackup
+
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:53]:
```suggestion
```

File: com.darhon.syncbackup.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+app-id: com.darhon.syncbackup
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: syncbackup
+build-options:
+    strip: true
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:7]:
```suggestion
```

You don't need this.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
These change are required to even build.

**Comment by sigulete** [comment]
Thanks. This is my first time packaging for flathub.
I have completed the changes. The only question is what is going to be the
final repository for my application, hence the url for the module
*syncbackup* may need to be modified.
I committed the changes, should I open a pull request again?

File: com.darhon.syncbackup.yml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    cleanup:
+      - /share/man
+
+  - name: syncbackup
+    buildsystem: cmake
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    sources:
+      - type: git
+        url: https://github.com/sigulete/flathub.git
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:41]: This repository doesn't contain the source code.

Also this REQUIRE a specific commit.

File: com.darhon.syncbackup.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        md5: 209f8326f5137d8817a6276d9577a2f1
+    cleanup:
+      - /share/man
+
+  - name: ksshaskpass
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: http://archive.ubuntu.com/ubuntu/pool/universe/k/ksshaskpass/ksshaskpass_5.22.5.orig.tar.xz
+        md5: 2d17696a462d4b1abc3f8b50811ba78c
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:31]: md5 is deprecated. Please use sha256.

File: com.darhon.syncbackup.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+modules:
+  - name: rsync
+    buildsystem: autotools
+    config-opts:
+      - --disable-zstd
+      - --disable-xxhash
+    sources:
+      - type: archive
+        url: http://archive.ubuntu.com/ubuntu/pool/main/r/rsync/rsync_3.2.3.orig.tar.gz
+        md5: 209f8326f5137d8817a6276d9577a2f1
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:22]: md5 is deprecated. Please use sha256.

File: com.darhon.syncbackup.yml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  - --filesystem=host
+modules:
+  - name: rsync
+    buildsystem: autotools
+    config-opts:
+      - --disable-zstd
+      - --disable-xxhash
+    sources:
+      - type: archive
+        url: http://archive.ubuntu.com/ubuntu/pool/main/r/rsync/rsync_3.2.3.orig.tar.gz
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:21]: why no using the official upstream?

File: com.darhon.syncbackup.yml [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        md5: 209f8326f5137d8817a6276d9577a2f1
+    cleanup:
+      - /share/man
+
+  - name: ksshaskpass
+    buildsystem: cmake
+    sources:
+      - type: archive
+        url: http://archive.ubuntu.com/ubuntu/pool/universe/k/ksshaskpass/ksshaskpass_5.22.5.orig.tar.xz
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:30]: why not using the official upstream?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.darhon.syncbackup.yml (hunk 8 above)
Comment (hfiguiere) [reply com.darhon.syncbackup.yml:41]: Just to clarify. This shoud point to the code of the software you are packaging. Not the flathub package repository.

File: com.darhon.syncbackup.yml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+      
+  - name: syncbackup_sh
+    buildsystem: simple
+    build-commands:
+      - install -D syncbackup.sh /app/bin/syncbackup.sh
+    sources:
+      - type: script
+        dest-filename: syncbackup.sh
+        commands: 
+          - syncbackup --ssh-askpass /app/bin/ksshaskpass
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:56]:
```suggestion
  - name: syncbackup
    buildsystem: cmake-ninja
    config-opts:
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
    post-install:
      - install -D syncbackup.sh /app/bin/syncbackup.sh
    sources:
      - type: git
        url: https://github.com/DarhonSoftware/syncBackup.git
        tag: v2.0.0
        commit: 0703b02ffb57a960d190bab89da647d94feccb83
      - type: script
        dest-filename: syncbackup.sh
        commands: 
          - syncbackup --ssh-askpass /app/bin/ksshaskpass
```

File: com.darhon.syncbackup.yml [hunk 14]
```diff
@@ -0,0 +1,57 @@
+app-id: com.darhon.syncbackup
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: syncbackup.sh
+finish-args:
+  - --socket=x11
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:7]:
```suggestion
  - --socket=fallback-x11
```

**Comment by hfiguiere** [comment]
bot, build com.darhon.syncbackup

**Comment by barthalion** [comment]
bot, build com.darhon.syncbackup

File: com.darhon.syncbackup.yml [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        commit: 17771f575050f4b21cc04522df0fc2d7f7013f44
+    cleanup:
+      - /share/man
+
+  - name: syncbackup
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      post-install:
+      - install -D syncbackup.sh /app/bin/syncbackup.sh
```
Comment (barthalion) [inline com.darhon.syncbackup.yml:44]: This is nested too much.

File: com.darhon.syncbackup.yml [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+        tag: v5.23.4
+        commit: 17771f575050f4b21cc04522df0fc2d7f7013f44
+    cleanup:
+      - /share/man
+
+  - name: syncbackup
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      post-install:
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:43]:
```suggestion
    post-install:
```

(this is the indentation problem mentionned below

File: com.darhon.syncbackup.yml [hunk 17]
```diff
@@ -0,0 +1,55 @@
+app-id: com.darhon.syncbackup
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: syncbackup.sh
+finish-args:
+  - --socket=x11
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:7]:
```suggestion
```

My suggestion was removing this and it wasn't.

File: com.darhon.syncbackup.yml [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+    sources:
+      - type: git
+        url: https://github.com/WayneD/rsync.git
+        tag: v3.2.3
+        commit: e94bad1c156fc3910f24e2b3b71a81b0b0bdeb70
+    cleanup:
+      - /share/man
+
+  - name: ksshaskpass
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.darhon.syncbackup.yml:30]:
```suggestion
    buildsystem: cmake-ninja
```

**Comment by hfiguiere** [comment]
bot, build com.darhon.syncbackup

**Comment by hfiguiere** [comment]
BTW you can also trigger build like this:


bot, build com.darhon.syncbackup

**Comment by DarhonSoftware** [comment]
bot, build com.darhon.syncbackup

**Pull request closed without merging**